### PR TITLE
feat(audit-service): multi tenant support

### LINKED ISSUE #1942: SaaS: Provision to get tenant wise logs - @sourceloop/audit-service
**Is your feature request related to a problem? Please describe.**
Right now we dont have any direct way to save and get logs on the basis of tenant
we should be able to get logs of only the particular tenant
tenant based filtering 
security using tenantGuard

### PR DESCRIPTION
**Pull request by Tyagi-Sunny**
## Description

multi tenant support

MIGRATION CHANGE:
migration-20240803165830- multi tenant support
migration-20231222043349- multi tenant support
migration-20231222043401- multi tenant support
migration-20231222043416- multi tenant support
migration-20210423140543- multi tenant support
migration-20230608151724- multi tenant support
migration-20230608151805- multi tenant support

## BREAKING

Fixes #1942 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Intermediate change (work in progress)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist:

- [ ] Performed a self-review of my own code
- [ ] npm test passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the style guide
- [ ] API Documentation in code was updated
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: services/audit-service/src/__tests__/integration/export-log.controller.unit.ts [hunk 1]
```diff
@@ -3,7 +3,7 @@ import {createStubInstance, expect} from '@loopback/testlab';
 import {AuthenticationBindings} from 'loopback4-authentication';
 import {FileStatusKey} from '../../enums/file-status-key.enum';
 import {OperationKey} from '../../enums/operation-key.enum';
-import {Job} from '../../models';
+import {Job} from '../../models/tenant-support';
```
[the remark this answers is not in the document]
Comment (yeshamavani) [reply services/audit-service/src/__tests__/integration/export-log.controller.unit.ts:6]: @Tyagi-Sunny

File: services/audit-service/src/controllers/audit.controller.ts [hunk 2]
```diff
@@ -186,7 +187,7 @@ export class AuditController {
         filterUsed: filter,
         status: FileStatusKey.PENDING,
         operation: OperationKey.QUERY,
-        tenantId: this.currentUser.tenantId,
+        // tenantId: this.currentUser.tenantId, --- check it on multi tenant support
```
[the remark this answers is not in the document]
Comment (yeshamavani) [reply services/audit-service/src/controllers/audit.controller.ts:190]: @Tyagi-Sunny

File: services/audit-service/src/__tests__/integration/export-log.controller.unit.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (Tyagi-Sunny) [reply services/audit-service/src/__tests__/integration/export-log.controller.unit.ts:6]: added

File: services/audit-service/src/component.ts [hunk 3]
```diff
@@ -110,8 +119,11 @@ export class AuditServiceComponent implements Component {
         JobRepository,
       ];
     }
-
-    this.models = [AuditLog, MappingLog, Job];
+    if (this.disableTenantGuard?.disable) {
+      this.models = [AuditLog, MappingLog, Job];
+    } else {
+      this.models = [TenantAuditLog, TenantMappingLog, TenantJob];
+    }
```
[the remark this answers is not in the document]
Comment (Tyagi-Sunny) [reply services/audit-service/src/component.ts:126]: ok

**Comment by yeshamavani** [comment]
This and https://github.com/sourcefuse/loopback4-audit-log/pull/103/files are inter-related 
separate the changes of @sourceloop/core in another PR 
as that has to be released first

File: services/audit-service/package.json [hunk 4]
```diff
@@ -72,8 +72,8 @@
     "@loopback/openapi-v3": "^10.0.0",
     "@loopback/repository": "^7.0.0",
     "@loopback/rest": "^14.0.0",
-    "@sourceloop/audit-log": "^7.0.0",
-    "@sourceloop/core": "^13.0.0",
+    "@sourceloop/audit-log": "file:/home/sunny.tyagi/Desktop/SourceFuse/2017/loopback4-audit-log/sourceloop-audit-log-7.0.0.tgz",
+    "@sourceloop/core": "file:/home/sunny.tyagi/Desktop/27_march_comments/loopback4-microservice-catalog/packages/core/sourceloop-core-13.0.0.tgz",
```
Comment (yeshamavani) [inline services/audit-service/package.json:76]: Dependency on https://github.com/sourcefuse/loopback4-microservice-catalog/pull/2063 and 
https://github.com/sourcefuse/loopback4-audit-log/pull/103

File: services/audit-service/src/__tests__/fixtures/dummy-application.ts [hunk 5]
```diff
@@ -17,6 +18,8 @@ export class DummyAuditServiceApplication extends BootMixin(
   constructor(options: ApplicationConfig = {}) {
     super(options);
     this.static('/', path.join(__dirname, '../public'));
+
+    this.component(AuditLogComponent);
```
Comment (yeshamavani) [inline services/audit-service/src/__tests__/fixtures/dummy-application.ts:22]: why ?

Comment (Tyagi-Sunny) [reply services/audit-service/src/__tests__/fixtures/dummy-application.ts:22]: as we also need to dynamically change models in audit log package

Comment (Tyagi-Sunny) [reply services/audit-service/src/__tests__/fixtures/dummy-application.ts]: changed

**Review by samarpan-b — APPROVED** [review]

File: services/audit-service/src/controllers/audit.controller.ts [hunk 6]
```diff
@@ -41,7 +41,8 @@ import {
   ColumnBuilderServiceBindings,
   ExportToCsvServiceBindings,
 } from '../keys';
-import {AuditLog, CustomFilter, Job, MappingLog} from '../models';
+import {CustomFilter} from '../models';
+import {AuditLog, Job, MappingLog} from '../models/tenant-support';
```
Comment (yeshamavani) [inline services/audit-service/src/controllers/audit.controller.ts:45]: check this

**Review by yeshamavani — CHANGES_REQUESTED** [review]

File: services/audit-service/src/controllers/audit.controller.ts (hunk 6 above)
Comment (Tyagi-Sunny) [reply services/audit-service/src/controllers/audit.controller.ts:47]: checked

**Review by yeshamavani — APPROVED** [review]

**Pull request merged**